hadoop-hdds/common/src/main/java/org/apache/hadoop/ozone/common/ChunkBuffer.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ *  with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package org.apache.hadoop.ozone.common;
+
+import org.apache.ratis.thirdparty.com.google.protobuf.ByteString;
+
+import java.nio.ByteBuffer;
+import java.util.function.Function;
+
+/** Buffer for a block chunk. */
+public interface ChunkBuffer {
+
+  /** Similar to {@link ByteBuffer#allocate(int)}. */
```

Comment:
NIT: can we move the allocate and wrap to ChunkBufferImplWithByteBuffer class too?

Reply:
It is standard to keep factory methods in the base class so that it can hide the implementations to users.  Examples are ByteBuffer.allocate(..), ByteString.copyFrom(..).

Reply:
Make sense. +1 and I will merge it shortly.